ansible/roles/pre-ansible/tasks/main.yml

```diff
@@ -9,6 +9,14 @@
   register: distro
   always_run: yes
 
+- name: Set the is_coreos fact
+  set_fact:
+    is_coreos: true
+  when: "'CoreOS' in distro.stdout"
+
+- include: coreos.yml
+  when: is_coreos
```

Comment:
oh man! All those comments and I was going to just merge.  Until the VERY LAST changeset! I believe you must initialize the fact. Else this `when: is_coreos` will crash on non-coreos systems.

You can fix only this, and I will merge hoping you'll update based on the other comments sometime in the future, or you can do as much or as little rework based on my review as you think reasonable.